[docs] initial QT documentation

Sjors opened the pull request:
I'll update this as I figure out how everything is tied together, but I think it's a useful enough start.

MarcoFalke commented on src/qt/README.md:
```diff
@@ -0,0 +1,90 @@
+
+## test
+
+Tests.
```
Not sure if listing folders whose description can be inferred from the name is useful.

Sjors replied:
It's a placeholder. I should explain how to run them, roughly what these tests (don't) cover, and perhaps point to some work in progress on UI tests.

jonasschnelli commented:
Not so sure.
I think design documents would be helpful, though may outdate quickly, especially if there are filenames.

Also, we have already a Qt Creator part in [OS X build docs](https://github.com/bitcoin/bitcoin/blob/41221126c85524a6dec34a4f44cb5b74b1f12e64/doc/build-osx.md#using-qt-creator-as-ide) (not sure if it belongs there).

promag commented:
I guess you should add `WIP: ` prefix to the title and meanwhile try to improve with useful documentation. Also not sure if that belongs to the source code or to some blog?..

fanquake commented:
This seems (so far) to mostly be a collection of documentation/links that are already available in the docs. I don't think there's much benefit in repeating build instructions, or other info that's already available elsewhere, as it just creates an opportunity for information to get out of sync.

I think @jonasschnelli has a point about the Qt Creator stuff that is just tacked onto the OS X build notes. If your going to do this, see if you can pull all the info together into a single file, rather than repeating and/or linking back to it.

Sjors commented:
My initial idea was to make it a quick introduction to the code in this directory with pointers to  more detailed information.

Given that QT is almost a standalone application, it could make sense to instead move all QT documentation to one place. In that case, would https://github.com/bitcoin-core/docs be more appropriate?

Either way, I don't think it should be too detailed, because then it won't be kept up to date. It should just be enough for a new developer - familiar with MVC, but not with QT - to get a sense of how the QT app is organized. More detailed descriptions can be inside individual source code files.

Some documentation could be avoided by adding more directories and using more consistent file naming, but that's future work, I think. 

@fanquake the build instructions are currently separated per OS (Unix is the most detailed). They don't specify how to build bitcoin-qt only (without compiling tests). I could add that there, or I could add links to build instructions for specific parts of the application. Maybe there should be a common build instructions file (pointing to each OS-specific file), which lists the various config flags and explains how to build specific targets?

laanwj commented:
I think the overview of files and directories is useful!

The building stuff I'd prefer to keep only in `doc` because it's different per platform, in some cases per compiler.

> They don't specify how to build bitcoin-qt only (without compiling tests)

Please don't encourage to skip building the tests.

I'd generalize that to "how to build individual executables with the automake build system", it's not Qt specific, the same could be used to build only the tests.

Sjors committed:
[docs] initial QT documentation, move Qt Creator instructions

Sjors commented:
I got rid of the build instructions, moved Qt Creator stuff over from OSX build and added a few hints.

I'll keep updating this PR as I wrap my head around all of QT's moving parts.

jb55 reviewed (approved):
ACK, useful overview for noobs like me! Subdirectory READMEs are a plus as well since GitHub renders them when browsing directories.

laanwj commented:
Looks good to me now. ACK https://github.com/bitcoin/bitcoin/pull/11761/commits/c8edc2c3cb79a1ae15622059004981c70f3775d8  (after you remove the WIP tag)

Sjors commented:
@laanwj WIP label removed
